### PR TITLE
HirIdify hir::Def

### PR DESCRIPTION
**Pull request by Zoxc**
cc @ljedrz
r? @oli-obk

### CODE REVIEW

**Comment by Zoxc** [comment]
This is ready for review now. cc @eddyb

File: src/librustc_resolve/lib.rs [hunk 1]
```diff
@@ -20,7 +21,9 @@ use rustc::hir::{self, PrimTy, Bool, Char, Float, Int, Uint, Str};
 use rustc::middle::cstore::CrateStore;
 use rustc::session::Session;
 use rustc::lint;
-use rustc::hir::def::*;
+use rustc::hir::def::{
+    self, PathResolution, CtorKind, CtorOf, NonMacroAttrKind, DefMap, ImportMap, ExportMap
+};
```
Comment (petrochenkov) [inline src/librustc_resolve/lib.rs]: Explicit definitions should shadow globs, so leaving the glob would still work.

**Review by eddyb — APPROVED** [review]
LGTM, r=me unless @petrochenkov, or anyone else, has any objections

**Review by petrochenkov — APPROVED** [review]

**Pull request merged**